### PR TITLE
watermark fallback

### PR DESCRIPTION
**Pull request by pldespaigne**
- watermark fix on develop #3650
- ~watermark fix on master #3734~
- ~watermark fix on production #3732~

### CODE REVIEW

**Review by phpgeekfr — APPROVED** [review]

File: libs/event/src/lib/components/player/player.component.ts [hunk 1]
```diff
@@ -47,20 +52,28 @@ export class EventPlayerComponent implements AfterViewInit, OnDestroy {
...
+      // if error is set, result will contain the error message
+      throw new Error(result);
     } else {
 
+      const userData = this.userQuery.getEntity(this.authQuery.userId);
+      const parameters: ImageParameters = {
+        auto: 'compress,format',
+        fit: 'crop',
+      };
+      const watermarkUrl = generateBackgroundImageUrl(userData.watermark, parameters);
```
Comment (phpgeekfr) [inline libs/event/src/lib/components/player/player.component.ts:72]: there will be a quick conflict on develop in the few days with `generateBackgroundImageUrl` (moved to mediaService).

Comment (pldespaigne) [reply libs/event/src/lib/components/player/player.component.ts:72]: alright, I will tackle that after your merge

File: apps/backend-functions/src/player.ts [hunk 2]
```diff
@@ -94,6 +95,54 @@ export const getPrivateVideoUrl = async (
...
+  let fileExists = false;
+
+  // if we have a ref we should assert that it points to an existing file
+  if (!!user.watermark) {
+    const ref = `public/users/${user.uid}/watermark/${user.uid}.svg`;
+    const file = admin.storage().bucket(bucketName).file(ref);
+    [fileExists] = await file.exists();
+  }
+
+  // if we don't have a ref OR if the file doesn't exists : we regenerate the Watermark
```
Comment (fritzschoff) [inline apps/backend-functions/src/player.ts:114]: why don't you concat this if statement with an else?

File: libs/event/src/lib/components/player/player.component.ts [hunk 3]
```diff
@@ -49,27 +47,34 @@ export class EventPlayerComponent implements AfterViewInit, OnDestroy {
...
+      const watermarkUrl = await this.mediaService.generateImgIxUrl(watermarkRef, parameters);
 
       const signedUrl = new URL(result);
       const expires = signedUrl.searchParams.get('exp');
       const timestamp = parseInt(expires, 10); // unix timestamp in seconds
       const millisecondTimestamp = timestamp * 1000; // js timestamp in milliseconds
       const refreshCountdown = millisecondTimestamp - Date.now();
 
-      this.timeout = setTimeout(() => window.location.reload(), refreshCountdown);
+      this.timeout = window.setTimeout(() => window.location.reload(), refreshCountdown);
```
Comment (fritzschoff) [inline libs/event/src/lib/components/player/player.component.ts:77]: why you need window here?

**Review by fritzschoff — APPROVED** [review]

File: libs/event/src/lib/components/player/player.component.ts (hunk 3 above)
Comment (pldespaigne) [reply libs/event/src/lib/components/player/player.component.ts:77]: without `window` TypeScript/linter was thinking I was using `NodeJS.Timoute` thus causing build errors

**Pull request merged**